services/frontend-service/src/ui/components/SideBar/SideBar.tsx

```diff
@@ -315,6 +315,15 @@ export const SideBar: React.FC<{ className?: string; toggleSidebar: () => void }
             action.action?.$case === 'createEnvironmentApplicationLock'
     );
     const [showSpinner, setShowSpinner] = useState(false);
+    const [dialogState, setDialogState] = useState({
+        showConfirmationDialog: false,
+    });
+    const cancelConfirmation = useCallback((): void => {
+        setDialogState({ showConfirmationDialog: false });
+    }, []);
+
+    const conflictingLocks = useLocksConflictingWithActions();
+    const hasLocks = conflictingLocks.environmentLocks.length > 0 || conflictingLocks.appLocks.length > 0;
```

Comment:
```suggestion
    const hasLocks = conflictingLocks.environmentLocks.length > 0 && conflictingLocks.appLocks.length > 0;
```
We can have only app locks or only env locks (and it would not be an error case). But here we say that we can have only env locks (for example) and we would get the dialog with conflicting locks

Reply:
Also the name of the variable should be different

Reply:
> here we say that we can have only env locks (for example) and we would get the dialog with conflicting locks

This is intended. If there is only one env lock, we show the dialog. Same if there is only one app lock.

Reply:
But if we only have one environment lock, we would be expecting the release dialog to only display the non-conflicting dialog. However, when we check `hasLocks` it only renders the non-conflicting dialog if the value is false. But `hasLocks` is set to true if we have either app locks or env locks, when I expect that it's set to true only if we have appLocks and envLocks. (services/frontend-service/src/ui/components/SideBar/SideBar.tsx line 420)

Reply:
Nevermind, the filter for actions is inside a filter for the environment locks/app locks. It is completely correct 👍